components/experimental/src/measure/single_unit_vec.rs

```diff
@@ -28,6 +28,21 @@ impl SingleUnitVec {
             SingleUnitVec::Multi(units) => &units[..],
         }
     }
+
+    pub(crate) fn push(&mut self, unit: SingleUnit) {
+        match self {
+            SingleUnitVec::Zero => *self = SingleUnitVec::One(unit),
+            SingleUnitVec::One(unit) => *self = SingleUnitVec::Two([*unit, *unit]),
+            SingleUnitVec::Two(units) => {
+                *self = SingleUnitVec::Multi(vec![units[0], units[1], unit]);
+            }
+            SingleUnitVec::Multi(units) => {
+                let mut units = core::mem::take(units);
+                units.push(unit);
+                *self = SingleUnitVec::Multi(units);
```

Comment:
I don't think the take and set is necessary, `units` should be a `&mut Vec<_>` that you can just push to